pytorch_lightning/plugins/training_type/tpu_spawn.py

```diff
@@ -173,16 +173,16 @@ def __load_weights_on_main_process(self) -> None:
 
         # load weights if not interrupted
         # TODO: check for trainer reference
-        if self.on_colab_kaggle and not model.trainer.testing:
+        if on_colab_kaggle() and not model.trainer.testing:
             self.load_spawn_weights(model)
 
-        self.lightning_module = model
+        self._model = model
 
     @property
     def xmp_spawn_kwargs(self):
         return {
-            "args": (self.lightning_module, trainer, self.mp_queue),
-            "nproc": len(self.parallel_devices),
+            "args": (self.lightning_module.trainer, ),
```

Comment:
I recently re-added the queue here as input, as it seems necessary to pass it into the process, 
the self.new_process also has to be adapted

Reply:
yep. `self.new_process` only accepts `trainer` arg as of now.